lsp-go.el

```diff
@@ -26,46 +26,77 @@
 
 (require 'lsp-mode)
 
-(defgroup lsp-gopls nil
```

Comment:
I used that prefix since `gopls` is the official LSP implementation for Go, it's the same for `rls`, as that is the official Rust implementation.

Reply:
Ah, good point. I still think we should leave the gopls namespace. Changing it is just config churn for everyone, and if someone down the line is unhappy with gopls and writes their own server, we won't have the same config problems we had gopls came along. @seagle0128 what do you think?

Reply:
Since `gopls` is the official langserver, I agree to rename the group to `lsp-go` for consistence. If another langserver is added, the names `lsp-go-[langserver]-xxx` are able to tell. But these servers belong to `lsp-go` group. It does make sense to me.